src/filebrowser/file-browser-dialog.cpp

```diff
@@ -1164,11 +1173,27 @@ void FileBrowserDialog::setFilesToBePasted(bool is_copy, const QStringList &file
 
 void FileBrowserDialog::onGetDirentsPaste()
 {
-    if (repo_id_to_be_pasted_from_ == repo_.id &&
-        current_path_ == dir_path_to_be_pasted_from_) {
-        seafApplet->warningBox(tr("Cannot paste files from the same folder"), this);
-        return;
+    if (repo_id_to_be_pasted_from_ == repo_.id) {
+        if (current_path_ == dir_path_to_be_pasted_from_) {
+            seafApplet->warningBox(tr("Cannot paste files from the same folder"), this);
+            return;
+        }
+
+        if (file_names_to_be_pasted_.isEmpty()) {
```

Comment:
No need to check this. If file_names_to_be_pasted_, the paste menu item would not be visible. 

See https://github.com/haiwen/seafile-client/blob/v6.1.0/src/filebrowser/file-table.cpp#L436

Reply:
But "Ctrl+v" will emit direntPaste() signal.